internal/rpc/jsonrpc/methods.go

```diff
@@ -3849,7 +3852,40 @@ func (s *Server) setTreasuryPolicy(ctx context.Context, icmd interface{}) (inter
 	}
 
 	err = w.SetTreasuryKeyPolicy(ctx, pikey, policy, ticketHash)
-	return nil, err
+	if err != nil {
+		return nil, err
+	}
+
+	// Update preference on VSP.
+	vspHost, err := w.VSPHostForTicket(ctx, ticketHash)
+	if err != nil {
+		return nil, err
```

Comment:
ticketHash could be nil here, and besides, I don't think we want to be returning errors for any tickets that were bought solo